File: Bit Manipulation/191.Number of 1 Bits.java
class Solution {
    public int hammingWeight(int n) {
        int ans=0;
        String s="";
        while(n>0){
            char c=(char)((n%2)+'0');
            s+=c;
            n/=2;
        }
        String rev= new StringBuilder(s).reverse().toString();
        for(int i=0;i<rev.length();i++){
            if(rev.charAt(i)=='1'){
                ans++;
            }
        }
        return ans;
    }
}

File: Bit Manipulation/191.Number of 1 Bits.cpp
class Solution {
public:
    int hammingWeight(int n) {
        int ans=0;
        // string s="";
        // while(n>0){
        //     s+=(n%2+'0'); //(n&1+'0')
        //     n/=2;//n>>=1;
        // }
        // for(int i=0;i<s.length();i++){
        //     if(s[i]=='1'){
        //         ans++;
        //     }
        // }
      
        while(n>0){
            if(n&1){
                ans++;
            }
            n/=2;//n>>=1;
        }
        return ans;
    }
};
